app/institutions/dashboard/-components/object-list/component.ts

```diff
@@ -0,0 +1,36 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+import InstitutionModel from 'ember-osf-web/models/institution';
+
+export interface ObjectListColumn {
+    name: string;
+    type?: 'link' | 'doi' | 'contributors';
+    valuePath?: string; // if no type is specified, this is the value to display
+    hrefValuePath?: string; // for links
+    linkText?: string; // for links
+}
```

Comment:
this column object seems a bit fraught with assumptions that would be better to make explicit... some suggestions (without changing the pattern)

- the constraints in the comments could be expressed more clearly with three separate interfaces (with a `type ObjectListColumn = X | Y | Z` that's essentially the same type you have here)
- using paths seems brittle and hard to follow... instead of `valuePath`, `hrefValuePath`, `linkText` as strings, could they be functions that take a `SearchResultModel`?

```ts
interface ValueColumn {
  name: string;
  getValue(searchResult: SearchResultModel): string;
}
interface LinkColumn {
  name: string;
  type: 'link';
  getHref(searchResult: SearchResultModel): string;
  getLinkText(searchResult: SearchResultModel): string;
}
interface SpecialColumn {
  name: string;
  type: 'doi' | 'contributors';
}
export type ObjectListColumn = ValueColumn | LinkColumn | SpecialColumn;
```

Reply:
Ahh, yea, I like this `getValue` pattern as well as the different and explicit interfaces here. Will implement it this way